.github/workflows/publish.yml

```diff
@@ -0,0 +1,77 @@
+name: Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish (e.g., 0.1.1, 0.2.0, 1.0.0)'
+        required: true
+        type: string
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v6
+        with:
+          node-version: 25
+          check-latest: true
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        run: npm i
+
+      - name: Biome
+        run: node --run biome:ci
+
+      - name: Typecheck
+        run: node --run typecheck
+
+      - name: ESLint
+        run: node --run eslint
+
+      - name: Prettier
+        run: node --run prettier:check
+
+      - name: Bundle
+        run: node --run build
+
+      - name: Build website
+        run: node --run build:website
+
+      - name: Install Playwright Browsers
+        run: npx playwright install chromium firefox
+
+      - name: Test
+        run: node --run test
+        timeout-minutes: 4
+
+      - name: Visual regression test
+        run: node --run visual
+
+      - name: Update version
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com"
+          npm version ${{ inputs.version }}
+
+      # https://docs.npmjs.com/trusted-publishers
+      - name: Publish to npm
+        run: npm publish
+
+      - name: Push commit and tag
+        run: git push origin main --follow-tags
+
+      - name: Create GitHub Release
+        run: gh release create v${{ inputs.version }} --title "v${{ inputs.version }}" --generate-notes
```

Comment:
We'll create immutable releases now, we'll have to write changelogs manually.
We can improve this in the future to automatically add changelogs from files, but this will do for now.

<img width="454" height="119" alt="Image" src="https://github.com/user-attachments/assets/96e0ef65-9174-473b-818e-43864d13caac" />